org.scala-ide.sdt.core.tests/src/org/scalaide/core/ui/OutlineModelTest.scala

```diff
@@ -0,0 +1,166 @@
+package org.scalaide.core.ui
+
+import org.scalaide.core.testsetup.TestProjectSetup
+import org.scalaide.core.CompilerTestUtils
+import org.junit.Test
+import org.junit.Assert
+import org.scalaide.ui.internal.editor.outline._
+import org.hamcrest.core.IsInstanceOf
+
+object OutlineModelTest extends TestProjectSetup("outline-model") {
```

Comment:
For this it is probably not an improvement but I just want to point it out: You do not need to use `TestProjectSetup` and therefore there is also no need for the `outline-model` project files. Instead, your object can mixin `CompilerSupportTests` and implement the `runTest` method as this:

```scala
private def runTest(str: String, f: RootNode => Unit): Unit = {
  import OutlineModelTest._
  withCompiler { comp =>
    val unit = mkScalaCompilationUnit(str)
    val rn = ModelBuilder.buildTree(comp, unit.sourceMap(str.toCharArray).sourceFile)
    f(rn)
  }
}
```

Only downside is that your package names would have to differ. Instead of just `pack` you could name them in the same way how the test method is named. This is necessary because the compilation units are generated and by using different package names we ensure that there are no clashes between type names etc.

But as I said only a small improvement if an improvement at all.